wrappers/leveldb/CMakeLists.txt

```diff
@@ -0,0 +1,32 @@
+include (ExternalProject)
+set(leveldb_PREFIX ${CMAKE_CURRENT_BINARY_DIR}/external/leveldb)
+ExternalProject_Add(
```

Comment:
It seems that the LevelDB people messed up their version naming. If you check the Release tab, the latest version is 1.22 and CMake build seems to be available. You probably did not find it because the versions before are names `v1.20` but later they are named just `1.22` (without the v). Could you check if it works by simply replacing the GIT_TAG in the previous command? Something Like:

```cmake
ExternalProject_Add(
    leveldb
    GIT_REPOSITORY https://github.com/google/leveldb.git
    GIT_TAG 1.22
    INSTALL_COMMAND ""  
    BUILD_BYPRODUCTS ${CMAKE_CURRENT_BINARY_DIR}/leveldb-prefix/src/leveldb-build/${CMAKE_STATIC_LIBRARY_PREFIX}leveldb${CMAKE_STATIC_LIBRARY_SUFFIX}
    CMAKE_ARGS -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER} -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER} -DCMAKE_CXX_FLAGS="-fPIC" -DLEVELDB_BUILD_TESTS=OFF -DLEVELDB_BUILD_BENCHMARKS=OFF
)
```

Reply:
First of all, thank you for the valuable reviews and your time. : )
I checked Cmake build available version 1.22 works so I replaced it.

Reply:
Can you check again if the changes were pushed? I can't see them in the current commit.

Reply:
Sorry, my mistake,  I just pushed it.

Reply:
Great! Thank you! I will check tomorrow what's causing the build problem before we can accept the PR. For now I will close all open Change Requests. Good job!

Reply:
After some read, I think it might be better starting using the new FetchContent module of CMake to add external projects. That seems to be a more clean interface and recommended way from now on. So I say we give it a try and make a few changes:

1. Replace the `include (ExternalProject)` by `include (FetchContent)` in line 1.
2. Before calling `FetchContent_Declare` we have to setup the variables manually, since it does not allow us to pass them the same way as `ExternalProject_Add`. So right after line 1 you can add
```cmake
set(LEVELDB_BUILD_TESTS OFF CACHE INTERNAL "") 
set(LEVELDB_BUILD_BENCHMARKS OFF CACHE INTERNAL "")
set(BUILD_SHARED_LIBS ON CACHE INTERNAL "")
```
The `BUILD_SHARED_LIBS` flag will build the shared library instead of static one. This has the advantage that all dependencies will be resolved automatically, so probably you won't have to worry about Snappy and other dependencies.

3. Replace the whole `ExternalProject_Add` call by:
```cmake
FetchContent_Declare(
  leveldb
  GIT_REPOSITORY https://github.com/google/leveldb.git
  GIT_TAG 1.22
)
```
4. Call `FetchContent_MakeAvailable(leveldb)` right after.
5. Then all we have to do is add the wrapper binary and link against leveldb with:
```cmake
add_library(leveldb_wrapper SHARED leveldb_wrapper.cpp)
target_link_libraries(leveldb_wrapper leveldb)
```
All other lines can be removed. Let's give this a try. I have the current commit saved in my local machine, so in case this does not work, we can go back to the original approach.

Reply:
I substitute `FetchContent` for `ExternalProject`
It becomes more clear, thank you for your valuable comments : )